setup.py

```diff
@@ -1,25 +1,172 @@
+            os.makedirs(target_dir)
+        r = requests.get(url, stream=True)
+        r.raise_for_status()
+
+        total = None
+        length = r.headers.get('Content-length')
+        if length is not None:
+            total = int(length) // chunk_size + 1
+
+        with open(target_path, 'wb') as f:
+            for chunk in tqdm(r.iter_content(chunk_size=chunk_size),
+                              desc="{}".format(filename),
+                              total=total,
+                              unit_scale=True, mininterval=15, unit=" chunks"):
+                sys.stdout.flush()
+                f.write(chunk)
+
+
+def _post_setup():
+    from multiprocessing import Pool
+    from nltk.downloader import download
+    download('punkt')
+
+    data_path = os.getenv('NLGEVAL_DATA', 'nlgeval/data')
+
+    path = 'nlgeval/word2vec/glove2word2vec.py'
+    if os.path.exists(path):
+        os.remove('nlgeval/word2vec/glove2word2vec.py')
+
+    downloads = []
+
+    if sys.version_info[0] == 2:
+        downloads.append(dict(
+            url='https://raw.githubusercontent.com/manasRK/glove-gensim/42ce46f00e83d3afa028fb6bf17ed3c90ca65fcc/glove2word2vec.py',
+            target_dir='nlgeval/word2vec'
+        ))
+    else:
+        downloads.append(dict(
+            url='https://raw.githubusercontent.com/robmsmt/glove-gensim/dea5e55f449794567f12c79dc12b7f75339b18ba/glove2word2vec.py',
+            target_dir='nlgeval/word2vec'
+        ))
+
+    setup_glove = not os.path.exists(os.path.join(data_path, 'glove.6B.300d.model.bin'))
+    if setup_glove:
+        downloads.append(dict(
+            url='http://nlp.stanford.edu/data/glove.6B.zip',
+            target_dir=data_path
+        ))
+
+    # Skip-thoughts data.
+    downloads.append(dict(
+        url='http://www.cs.toronto.edu/~rkiros/models/dictionary.txt',
+        target_dir=data_path
+    ))
+    downloads.append(dict(
+        url='http://www.cs.toronto.edu/~rkiros/models/utable.npy',
+        target_dir=data_path
+    ))
+    downloads.append(dict(
+        url='http://www.cs.toronto.edu/~rkiros/models/btable.npy',
+        target_dir=data_path
+    ))
+    downloads.append(dict(
+        url='http://www.cs.toronto.edu/~rkiros/models/uni_skip.npz',
+        target_dir=data_path
+    ))
+    downloads.append(dict(
+        url='http://www.cs.toronto.edu/~rkiros/models/uni_skip.npz.pkl',
+        target_dir=data_path
+    ))
+    downloads.append(dict(
+        url='http://www.cs.toronto.edu/~rkiros/models/bi_skip.npz',
+        target_dir=data_path
+    ))
+    downloads.append(dict(
+        url='http://www.cs.toronto.edu/~rkiros/models/bi_skip.npz.pkl',
+        target_dir=data_path
+    ))
+
+    # multi-bleu.perl
+    downloads.append(dict(
+        url='https://raw.githubusercontent.com/moses-smt/mosesdecoder/b199e654df2a26ea58f234cbb642e89d9c1f269d/scripts/generic/multi-bleu.perl',
+        target_dir='nlgeval/multibleu'
+    ))
+
+    pool = Pool(len(downloads))
```

Comment:
Seems like too many files are being downloaded in parallel from http://www.cs.toronto.edu.